test/test_linalg.py

```diff
@@ -674,6 +675,122 @@ def test_norm_fastpaths(self, device):
         expected = torch.pow(x.pow(3).abs().sum(1), 1.0 / 3.0)
         self.assertEqual(result, expected)
 
+    @skipCPUIfNoLapack
+    @skipCUDAIfNoMagma
+    @unittest.skipIf(not TEST_NUMPY, "NumPy not found")
+    @dtypes(torch.double, torch.float)
+    def test_eig_basic(self, device, dtype):
```

Comment:
Can this test and the next two be simplified by enumerating test cases and just comparing with NumPy? 

The test cases could include discontiguous tensors. 

I think the out variant of eig could also be tested at the same time. Curious to hear your thoughts on this.